esphome/components/speaker/__init__.py

```diff
@@ -81,9 +81,6 @@ async def speaker_play_action(config, action_id, template_arg, args):
 automation.register_action("speaker.stop", StopAction, SPEAKER_AUTOMATION_SCHEMA)(
     speaker_action
 )
-automation.register_action("speaker.finish", FinishAction, SPEAKER_AUTOMATION_SCHEMA)(
-    speaker_action
-)
```

Comment:
Lets leave this in so its not a config breaking change and just change the class it uses.
```suggestion
automation.register_action("speaker.finish", StopAction, SPEAKER_AUTOMATION_SCHEMA)(
    speaker_action
)
```

Reply:
@jesserockz, the Stop and finish are 2 different thinks. The first stops the speaker directly while finish should wait until the everything is send out and then close the speaker.
Changing the finish action by doing the full stop as you suggested is as a breaking change as removing. it is worse, imho, because the end-user will think it will wait and found that the end of the audio stream is not being played anymore after calling finish. By removing the action totally will make the end-user investigate what is the problem and will found out that he does not need to call finish anymore instead he could wait until the speaker is stopped playing. 

One solution could be that we make `finish` a wait action until the speaker stops.